internal/provider/models/api_token.go

```diff
@@ -0,0 +1,75 @@
+package models
+
+import (
+	"context"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients/iam"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/schemas"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// ApiToken describes the data source data model.
+type ApiToken struct {
+	Id                 types.String `tfsdk:"id"`
+	Name               types.String `tfsdk:"name"`
+	Description        types.String `tfsdk:"description"`
+	ShortToken         types.String `tfsdk:"short_token"`
+	Type               types.String `tfsdk:"type"`
+	StartAt            types.String `tfsdk:"start_at"`
+	EndAt              types.String `tfsdk:"end_at"`
+	CreatedAt          types.String `tfsdk:"created_at"`
+	UpdatedAt          types.String `tfsdk:"updated_at"`
+	CreatedBy          types.Object `tfsdk:"created_by"`
+	UpdatedBy          types.Object `tfsdk:"updated_by"`
+	ExpiryPeriodInDays types.Int64  `tfsdk:"expiry_period_in_days"`
+	LastUsedAt         types.String `tfsdk:"last_used_at"`
+	Roles              types.Set    `tfsdk:"roles"`
+	Token              types.String `tfsdk:"token"`
```

Comment:
will prob need to split up this into ApiTokenResource and ApiTokenDataSource similar to Deployment since `Token` will only show up in resource
- will need to figure out how to do `rotate` and `updating roles`